lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -17,28 +17,22 @@ class EjsonSecretProvisioner
     EJSON_SECRETS_FILE = "secrets.ejson"
     EJSON_KEYS_SECRET = "ejson-keys"
 
-    def initialize(namespace:, context:, template_dir:, logger:, prune: true)
+    def initialize(namespace:, context:, template_dir:, logger:)
       @namespace = namespace
       @context = context
       @ejson_file = "#{template_dir}/#{EJSON_SECRETS_FILE}"
       @logger = logger
-      @prune = prune
       @kubectl = Kubectl.new(
         namespace: @namespace,
         context: @context,
         logger: @logger,
         log_failure_by_default: false,
-        output_is_sensitive: true # output may contain ejson secrets
+        output_is_sensitive_default: true # output may contain ejson secrets
       )
     end
 
-    def secret_changes_required?
-      File.exist?(@ejson_file) || managed_secrets_exist?
-    end
-
-    def run
-      create_secrets
-      prune_managed_secrets if @prune
+    def resources
+      @resources ||= File.exist?(@ejson_file) ? create_secrets : []
```

Comment:
Seems like File.exist?(@ejson_file)` is an implementation detail leaking. ` Maybe `create_secrets` should return `[]` or the `exist?` call should be wrapped in a function

Reply:
`#with_decrypted_ejson` already does that check, so it's somewhat redundant. I did `with_decrypted_ejson do .... end || []`.

Reply:
I wrote this in the distant past of last week 😅iirc I chose this approach because some of the log assertions were failing in cases when e.g. there's no matching key. I'll have another pass and try to tidy this up.